rust/lance-namespace/src/dir.rs

```diff
@@ -0,0 +1,674 @@
+            .map(|m| m.iter().map(|(k, v)| (k.clone(), v.clone())).collect())
+            .unwrap_or_default();
+        
+        Ok(Schema::new_with_metadata(fields, metadata))
+    }
+    
+    fn convert_json_arrow_type_to_arrow(&self, json_type: &JsonArrowDataType) -> Result<DataType> {
+        match json_type.r#type.to_lowercase().as_str() {
+            "null" => Ok(DataType::Null),
+            "bool" | "boolean" => Ok(DataType::Boolean),
+            "int8" => Ok(DataType::Int8),
+            "uint8" => Ok(DataType::UInt8),
+            "int16" => Ok(DataType::Int16),
+            "uint16" => Ok(DataType::UInt16),
+            "int32" => Ok(DataType::Int32),
+            "uint32" => Ok(DataType::UInt32),
+            "int64" => Ok(DataType::Int64),
+            "uint64" => Ok(DataType::UInt64),
+            "float32" => Ok(DataType::Float32),
+            "float64" => Ok(DataType::Float64),
+            "utf8" => Ok(DataType::Utf8),
+            "binary" => Ok(DataType::Binary),
+            _ => Err(NamespaceError::InvalidConfiguration(format!("Unsupported Arrow type: {}", json_type.r#type))),
+        }
+    }
+}
+
+#[async_trait]
+impl LanceNamespace for DirNamespace {
+    async fn create_namespace(&self, _request: CreateNamespaceRequest) -> Result<CreateNamespaceResponse> {
+        Err(NamespaceError::NotSupported(
+            "Directory namespace only contains a flat list of tables and does not support creating namespaces".to_string()
+        ))
+    }
+    
+    async fn list_namespaces(&self, _request: ListNamespacesRequest) -> Result<ListNamespacesResponse> {
+        Err(NamespaceError::NotSupported(
+            "Directory namespace only contains a flat list of tables and does not support listing namespaces".to_string()
+        ))
+    }
+    
+    async fn describe_namespace(&self, _request: DescribeNamespaceRequest) -> Result<DescribeNamespaceResponse> {
+        Err(NamespaceError::NotSupported(
+            "Directory namespace only contains a flat list of tables and does not support describing namespaces".to_string()
+        ))
+    }
+    
+    async fn drop_namespace(&self, _request: DropNamespaceRequest) -> Result<DropNamespaceResponse> {
+        Err(NamespaceError::NotSupported(
+            "Directory namespace only contains a flat list of tables and does not support dropping namespaces".to_string()
+        ))
+    }
+    
+    async fn namespace_exists(&self, _request: NamespaceExistsRequest) -> Result<()> {
+        Err(NamespaceError::NotSupported(
+            "Directory namespace only contains a flat list of tables and does not support namespace existence checks".to_string()
+        ))
+    }
+    
+    async fn list_tables(&self, request: ListTablesRequest) -> Result<ListTablesResponse> {
+        self.validate_root_namespace_id(&request.id)?;
+        
+        let mut tables = Vec::new();
+        
+        let entries = self.operator.list("").await?;
```

Comment:
this does not handle object store that does not have folder concept. Check the python and java implementation for the fix.